storage_engine/page_store/README.md

```diff
@@ -0,0 +1,90 @@
+# PageStore
+
+PageStore module is for storing pages in a storage file, and perform CRUD operations on the pages.
+
+## What is a page?
+
+A page is a storage unit of constant size (this constant size is `page_length`).
+
+Each page can store data of size less or equal to `page_length`.
+
+### Terminologies
+
+- `page_length`: The size of a page.
+- `page_index`: The unique serial index of a page in the storage.
+- `page_payload`: The data stored in a page. (excluding page header)
+- `page_payload_length`: The size of the data payload of a page. (This is the page header)
+
+### Layout of a page
+
+```txt
+|--------------------------------------|
+| page_payload_length as little endian   | <- Page header |
+| -------------------------------------- |
+| page_payload                           | <- Page data   |
+| -------------------------------------- |
+```
+
+#### Size of `page_payload_length` little endian byte representation
+
+The size this page header depends on the constant page length. We have made it adaptive to make storage more efficient.
```

Comment:
Bad sentence

Reply:
Updating to

```
`page_payload_length` is an unsigned integer stored as little-endian bytes. Since the maximum value of `page_payload_length` is always less than `page_length`, we have implemented adaptive byte representation to save storage space & traversal time.
```

Reply:
Table of Adaptive types & byte representation sizes

| Page length             | Unsigned integer type | Bytes required |
| ----------------------- | --------------------- | -------------- |
| <= 255                  | u8                    | 1              |
| <= 65535                | u16                   | 2              |
| <= 4294967295           | u32                   | 4              |
| <= 18446744073709551615 | u64                   | 8              |